app/models/user.rb

```diff
@@ -1,5 +1,15 @@
 class User < ActiveRecord::Base
   has_secure_password
+  has_many :clients_coaches, class_name: 'CoachClient',
+                           foreign_key: 'client_id',
+                           dependent: :destroy
+  has_many :coach_clients, class_name: 'CoachClient',
+                           foreign_key: 'coach_id',
+                           dependent: :destroy
+  has_many :coaches, through: :clients_coaches, source: :coach
+  has_many :clients, through: :coach_clients, source: :client
 
-  validates_uniqueness_of :email
+  def coach
+    self.coaches.first
```

Comment:
Tbh I don't know the requirements (for example if a user can have two or more coaches at the same time) but maaaaybe we'd like to return active coach here